Backpack-SwiftUI/Calendar/Classes/BPKCalendar.swift

```diff
@@ -40,6 +43,7 @@ public struct BPKCalendar<DayAccessoryView: View>: View {
     private var accessoryAction: ((Date) -> CalendarMonthAccessoryAction?)?
     private var initialMonthScroll: MonthScroll?
     private let monthHeaderDateFormatter: DateFormatter
+    private let calendarSelectionHandler: CalendarSelectionHandler
```

Comment:
I think this introduces a bit of uncertainty on the API, should this be a property of the selection type itself? 
For example, if i configure the calendar to be single selection only, why would it ask me to provide a way to handle range selections? the API should not allow me to get into inconsistent behaviour or confusion. does that make sense?

Reply:
on second thought, I'm not really sure this is a feature the calendar would want to expose. bpk components should define the behaviours of the components, only opening for flexibility when the behaviour is to be defined by the client. for the calendar it doesnt seem like this is what we'd want.
I see how this makes it 'easier' for clients to put new functionality in, but sometimes convenience opens up the components for potentially misbehaving (as in, doing something it's not intended to do).

Reply:
I think what we'd want to do on this case is make the code flexibe to allow different options, but to restrict the API of the compoennt for the clients to 'select' which behaviour they want, rather than letting them define it.